cmd/frontman/main.go

```diff
@@ -138,244 +105,329 @@ sudo sysctl -w net.ipv4.ping_group_range="0   2147483647"`
-		if serviceInstallPtr != nil && *serviceInstallPtr || serviceInstallUserPtr != nil && *serviceInstallUserPtr != "" {
-			fmt.Println(" ****** Before start you need to set 'hub_url' config param at ", *cfgPathPtr)
-		} else {
-			fmt.Println("Missing input file flag(-i) or hub_url param in config")
-			flag.PrintDefaults()
-			return
+	systemService, err := getServiceFromFlags(fm, cfgPathPtr, serviceInstallUserPtr)
+
+	if !service.Interactive() {
+		// we are running under OS service manager
+		err = systemService.Run()
+
+		if err != nil {
+			log.Fatal(err.Error())
 		}
+
+		os.Exit(0)
 	}
 
-	var output *os.File
+	flagServiceUninstall(systemService, *serviceUninstallPtr)
 
-	if inputFilePtr != nil && *inputFilePtr != "" {
+	// check some incompatible flags
+	if serviceInstallUserPtr != nil && *serviceInstallUserPtr != "" ||
+		serviceInstallPtr != nil && *serviceInstallPtr {
+		if *inputFilePtr != "" {
+			fmt.Println("Input file(-i) flag can't be used together with service install(-s) flag")
+			os.Exit(1)
+		}
 
-		if outputFilePtr == nil || *outputFilePtr == "" {
-			*outputFilePtr = "./results.out"
+		if *outputFilePtr != "" {
+			fmt.Println("Output file(-o) flag can't be used together with service install(-s) flag")
+			os.Exit(1)
 		}
+	}
 
-		if *outputFilePtr != "-" {
+	flagServiceInstall(systemService, fm, serviceInstallUserPtr, serviceInstallPtr)
+	flagDaemonizeMode(*daemonizeModePtr)
 
-			if _, err := os.Stat(*outputFilePtr); os.IsNotExist(err) {
-				dir := filepath.Dir(*outputFilePtr)
-				if _, err := os.Stat(dir); os.IsNotExist(err) {
-					err = os.MkdirAll(dir, 0644)
-					if err != nil {
-						log.WithError(err).Errorf("Failed to create the output file directory: '%s'", dir)
-					}
-				}
-			}
+	output := flagInputOutput(*inputFilePtr, *outputFilePtr, *oneRunOnlyModePtr)
 
-			mode := os.O_WRONLY | os.O_CREATE
+	interruptChan := make(chan struct{})
+	flagOneRunOnlyMode(fm, *oneRunOnlyModePtr, *inputFilePtr, output, interruptChan)
 
-			if *oneRunOnlyModePtr {
-				mode = mode | os.O_TRUNC
-			} else {
-				mode = mode | os.O_APPEND
-			}
+	// no any flag resulted in os.Exit
+	// so lets use the default continuous run mode
+	doneChan := make(chan struct{})
+	go func() {
+		fm.Run(*inputFilePtr, output, interruptChan)
+		doneChan <- struct{}{}
+	}()
 
-			output, err = os.OpenFile(*outputFilePtr, mode, 0644)
-			defer output.Close()
+	select {
+	case sig := <-sigc:
+		log.Infof("Got %s signal. Finishing the batch and exit...", sig.String())
+		interruptChan <- struct{}{}
+		os.Exit(0)
+	case <-doneChan:
+		return
+	}
+}
 
-			if err != nil {
-				log.WithError(err).Fatalf("Failed to open the output file: '%s'", *outputFilePtr)
+func flagVersion(versionFlag bool) {
+	if versionFlag {
+		fmt.Printf("frontman v%s released under MIT license. https://github.com/cloudradar-monitoring/frontman/\n", version)
+		os.Exit(0)
+	}
+}
+
+func flagPrintConfig(printConfig bool, fm *frontman.Frontman) {
```

Comment:
first I made all the methods starts with handle, like `handlePrintConfigFlag` then renamed to `flagPrintConfig` to make it looks better. But I agree. It will be better to name `handleFlagPrintConfig`

Reply:
renamed